app_backend/app/routers/favorites.py

```diff
@@ -34,11 +44,17 @@ async def toggle_favorite(resp: Response, favorite: FavoriteFood):
         resp.status_code = status.HTTP_400_BAD_REQUEST
         return {"err": "Empty food_id"}
 
-    existing_food_state = True  # TODO: get food_id's state here
-    if existing_food_state == favorite.state:
+    favorites = user.favorites
+    if (favorite.food_name in favorites) == favorite.state:
         resp.status_code = status.HTTP_200_OK
         return {"msg": "No row is affected."}
-    # TODO: Update food_id here
-    # toggle food_id
+    else:
+        if favorite.state:
```

Comment:
Not sure if I am understanding this bit correctly, how does it toggle from already favorite to not favorite? It seems like you have a list of favorite food names. I like this -- maybe just get rid of states? 

When toggling from favorite to not favorite remove from the list?

Reply:
Oh, I forgot to remove them from the list. Will do it now.